src/bromwhich_contour.jl

```diff
@@ -0,0 +1,138 @@
+### Implement Laplace transform along a hyperbola contour ###
+
+# 1.
+# Weideman, J., and L. Trefethen.
+# Parabolic and hyperbolic contours for computing the Bromwich integral.
+# Mathematics of Computation 76.259 (2007): 1341-1356.
+
+# 2.
+# Liemert, André, and Alwin Kienle.
+# Application of the Laplace transform in time-domain optical spectroscopy and imaging.
+# Journal of biomedical optics 20.11 (2015): 110502.
+
+### adaptive contour for each time point (can't precompute f(s) or sk)
+
+# parameterized hyperbola contour
+# parameters are given by ref [2] which are given in increased precision from calculations given in [1]
+function s(θ, N, t)
+    μ = 4.492075287 * N / t
+    ϕ = 1.172104229 
+    return μ + im * μ * sinh(θ + im * ϕ)
+end
+# derivitive of hyperbola contour
+function ds(θ, N, t)
+    μ = 4.492075287 * N / t
+    ϕ = 1.172104229
+    return im * μ * cosh(θ + im * ϕ)
+end
+
+# compute the laplace transform along a hyperbola contour fixed time point
+"""
+    hyperbola(f::Function, t::AbstractFloat; N::Int = 16)
+
+Evaluate the inverse Laplace transform of `f` at the point `t` by approximating the Bromwhich integral with a hyperbola contour.
+A real valued time-domain signal is assumed with the integral being calculated by a computable series applying the midpoint rule.
+
+The contour is defined by several parameters. μ controls the extreme nodes with a larger parameter moving the outlier nodes further into left half plane.
+h is the uniform node spacing with N being the number of nodes on the contour. The parameters depend on both t and N such that the function must be computed N times for each t.
+This is very inefficient especially if f is computational expensive, so this approach is recommended when only a few values of t are needed.
+However, this method will provide more accurate results at the cost of computational time compared to fixed contour approaches.
+
+The number of nodes, N, defaults to 16. The optimal N is case dependent and arbitrarily increasing N will not result in more accurate results. Smaller values of N will be faster.
+
+Bromwhich contour approaches should only be applied for t > 0. We are aiming to optimize the convergence as N -> inf so our parameters are ~ N / t.
+In other words we want the Bromwhich integral to converge rapidly, and this can be done by starting and ending our integration path in the left hand plane causing exp(st) to decay.
+If t = 0, the exponential factor does not cause good convergence. If you want to evaluate for t = 0, you should try a small positive value like 1e-30.
+
+# Example
+
+```jldoctest
+julia> InverseLaplace.hyperbola(s -> 1/(s + 1), 2.0)
+0.13533528323665164
+"""
+function hyperbola(f::Function, t::AbstractFloat; N = 16)
+    a =  zero(Complex{eltype(t)})
+    N = convert(eltype(t), N)
```

Comment:
Probably not needed to convert types here as it will be type stable without this if `N` is an Int. We should limit `N` to be an integer as line 57 will not be accurate if it's not.

Reply:
Yes. For a number of reasons,  you don't want to convert `N`.

Reply:
Yes. For a number of reasons,  you don't want to convert `N`.